src/AEG/CutNode.ts

```diff
@@ -115,32 +115,65 @@ export class CutNode {
     }
 
     /**
-     * Method that checks whether the given node can be inserted into this cut
-     * at a given point without overlapping any bounding boxes.
+     * Checks whether the incoming node can be inserted into this CutNode
+     * at a given point without overlapping any boundaries.
      * @param incomingNode The node to be inserted.
-     * @param insertionPoint The point at which the node must be inserted
      * @returns True, if the node can be inserted. Else, false
      */
-    public canInsert(incomingNode: AtomNode | CutNode, insertionPoint: Point): boolean {
-        //TO BE IMPLEMENTED??
-
-        const isValid = true;
-
+    public canInsert(incomingNode: AtomNode | CutNode): boolean {
```

Comment:
You never check to make sure an atom or cut is INSIDE the this cut, you just make sure they don't overlap.